views/js/runner/helpers/progress.js

```diff
@@ -0,0 +1,356 @@
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2018 (original work) Open Assessment Technologies SA ;
+ */
+
+/**
+ * Compute the current progress in the Test Runner
```

Comment:
Is this _helper_ used by other component than the progress plugin ?

Reply:
For now, this helper is only used by the progress plugin.

Reply:
I was wondering why the stats of the tests aren't sufficient to compute the progress : 
1. if this is the case and this helper is only about formatting the stats, then it should be moved inside the plugin folder too ?
2. OR  It fill a gap in the testMap that contains inaccurate data (ie. `viewed` values are not up to date), in that situation shouldn't we update directly the testMap ?

Reply:
Correct, this is for the first case: the regular stats are not sufficient to compute the progress.
I placed this helper as a common helper because I was wondering if other plugins could benefit of it. If you think this is too much specific to the progress bar plugin and useless for others, I can move it to the plugin folder.

Reply:
My main concern is that it looks like the place the progression is computed, globally and can lead to misunderstanding. 
It's safer in my opinion to move it into the plugin folder. If we have a case that needs it we will generalize it.

Reply:
ACK, I will move it